wire-ios-sync-engine/Source/Use cases/DisableAnalyticsUseCase.swift

```diff
@@ -29,21 +29,35 @@ public protocol DisableAnalyticsUseCaseProtocol {
 
 /// Concrete implementation of the DisableAnalyticsUseCaseProtocol.
 /// This struct is responsible for disabling analytics sharing.
-public struct DisableAnalyticsUseCase: DisableAnalyticsUseCaseProtocol {
+struct DisableAnalyticsUseCase<UserSession>: DisableAnalyticsUseCaseProtocol
+where UserSession: DisableAnalyticsUseCaseUserSession {
 
-    private let analyticsManager: AnalyticsManagerProtocol
+    private let sessionManager: AnalyticsManagerProviding
+    private let userSession: UserSession
 
-    /// Initializes a new instance of DisableAnalyticsUseCase.
+    /// Initializes a new instance of `DisableAnalyticsUseCase`.
     ///
-    /// - Parameter analyticsManager: The analytics manager to use for disabling tracking.
-    public init(analyticsManager: AnalyticsManagerProtocol) {
-        self.analyticsManager = analyticsManager
+    /// - Parameters:
+    ///   - sessionManager: The session manager that conforms to `AnalyticsManagerProviding` for managing analytics sessions.
+    ///   - didDisableAnalytics: A closure to be executed after analytics have been disabled.
+    init(
+        sessionManager: AnalyticsManagerProviding,
+        userSession: UserSession
```

Comment:
Consider renaming `userSession` here and else where to something like `analyticsSessionProvider` as we don't really care about the `userSession` functionality - only that it provides the analytics session. It could be any object.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/97b486e3536ba60882aa7ac9a80cea7c6f76e7de